app/controllers/users_controller.rb

```diff
@@ -25,4 +37,12 @@ def user_params
     params.require(:user).permit(:email, :password, :full_name)
   end
   
+  def handle_invitation
+    if params[:invitation_token].present?
+      invitation = Invitation.find_by(token: params[:invitation_token])
+      @user.follow(invitation.inviter)
+      invitation.inviter.follow(@user)
+      invitation.update_column(:token, nil)
```

Comment:
You could refactor `.update_column(:token, nil)` this part to the Invitation model and call it as `remove_token`